### PR TITLE
modify several module's SettingDefinitionProvider to support multi-lingual

### PR DESCRIPTION
**Pull request by yinchang0626**
configure embedded resources by wildcard and rebase from latest dev  branch
@maliming please help to add to Milestone 1.2
#2332

### CODE REVIEW

**Comment by hikalkan** [comment]
Thank you for your great contribution.
One problem I see that `DisplayName:Abp.Localization.DefaultLanguage` is added to the ValidationResource which is irrelevant.

@maliming you can merge it after reviewing, then create an `LocalizationModuleResource` (or find a better naming) and move this definition to there.

**Comment by yinchang0626** [comment]
@hikalkan 
You are right,
i think create a new resourceName in Volo.Abp.Localization is better,
Now Volo.Abp.Localization module just have  AbpValidationResource and maybe that should be move to Volo.Abp.Validation.
I can try to do it in new pull request

**Comment by maliming** [comment]
@yinchang0626 You can continue to commit new code on this branch without reopening pr.

**Comment by yinchang0626** [comment]
@hikalkan @maliming 
I am going to move `AbpValidationResource` from `Volo.Abp.Localization` to `Volo.Abp.Validation` Module,
```csharp
namespace Volo.Abp.Localization.Resources.AbpValidation
{
    //TODO: Move to Volo.Abp.Validation!

    [LocalizationResourceName("AbpValidation")]
    public class AbpValidationResource
    {
        
    }
}

```
and let `AbpValidationModule` Depends `AbpValidationModule`
```csharp
    [DependsOn(
        typeof(AbpLocalizationModule)
        )]
    public class AbpValidationModule : AbpModule
    {
        public override void PreConfigureServices(ServiceConfigurationContext context)
        {
            Configure<AbpVirtualFileSystemOptions>(options =>
            {
                options.FileSets.AddEmbedded<AbpValidationModule>();
            });

            Configure<AbpLocalizationOptions>(options =>
            {
                options.Resources
                    .Add<AbpValidationResource>("en")
                    .AddVirtualJson("/Volo/Abp/Localization/Resources/AbpValidation");
            });
......
```
but after moved,will cause breaking change,because of all other modules and template(app,module)  is depend `Volo.Abp.Localization` for 
```csharp
            Configure<AbpLocalizationOptions>(options =>
            {
                options.Resources
                    .Add<IdentityResource>("en")
                    .AddBaseTypes(
                        typeof(AbpValidationResource)
                    ).AddVirtualJson("/Volo/Abp/Identity/Localization");
            });
```

so exised module need to modify to
image
Could I change it like this for all modules?
or just add a `LocalizationModuleResource` for `DisplayName:Abp.Localization.DefaultLanguage` in `Volo.Abp.Localization`??

File: framework/src/Volo.Abp.Emailing/Volo/Abp/Emailing/AbpEmailingModule.cs [hunk 1]
```diff
@@ -30,6 +31,15 @@ public override void ConfigureServices(ServiceConfigurationContext context)
                 options.FileSets.AddEmbedded<AbpEmailingModule>();
             });
 
+            Configure<AbpLocalizationOptions>(options =>
+            {
+                options.Resources
+                    .Add<EmailingResource>("en")
+                    .AddBaseTypes(
+                        typeof(EmailingResource)
```
Comment (maliming) [inline framework/src/Volo.Abp.Emailing/Volo/Abp/Emailing/AbpEmailingModule.cs]: Why does EmailingResource inherit EmailingResource(`AddBaseTypes`)?

**Comment by wakuflair** [comment]
How about defining setting localization resources in just one resource, (e.g., "SettingResource")?
In that case,  it's convenient to show the localizaed string for all the settings.
(It is what [my module](https://github.com/wakuflair/Abp.SettingManagement.Mvc.UI) is doing.)

File: framework/src/Volo.Abp.Emailing/Volo/Abp/Emailing/AbpEmailingModule.cs (hunk 1 above)
Comment (yinchang0626) [reply framework/src/Volo.Abp.Emailing/Volo/Abp/Emailing/AbpEmailingModule.cs]: thinks,I fixed it

**Comment by yinchang0626** [comment]
@wakuflair 
of course,put all in one resource is a way and works fine,
but maybe add new module in future,then have to modify "SettingResource",I think  module should not be influenced by others module.
 So setting value is defined in independent module,that "DisplayName" and "Description" should be follow it.

**Comment by wakuflair** [comment]
The future module could inherit "SettingResource", or just add its own resources by using the `AddVirtualJson` method.
It's very reasonable for the usage of settings.

**Comment by yinchang0626** [comment]
@wakuflair 
Any way, thanks for your [module](https://github.com/wakuflair/Abp.SettingManagement.Mvc.UI)
`SettingDefinition` already have a Constructor which with `displayName`,`description` params.
Why not define it in first(framework) ?
```csharp
    internal class EmailSettingProvider : SettingDefinitionProvider
    {
        public override void Define(ISettingDefinitionContext context)
        {
            context.Add(
                new SettingDefinition(EmailSettingNames.Smtp.Host, "127.0.0.1", L("DisplayName:Abp.Mailing.Smtp.Host"), L("Description:Abp.Mailing.Smtp.Host")),
                new SettingDefinition(EmailSettingNames.Smtp.Port, "25", L("DisplayName:Abp.Mailing.Smtp.Port"), L("Description:Abp.Mailing.Smtp.Port")),
                new SettingDefinition(EmailSettingNames.Smtp.UserName, displayName: L("DisplayName:Abp.Mailing.Smtp.UserName"), description: L("Description:Abp.Mailing.Smtp.UserName")),
                new SettingDefinition(EmailSettingNames.Smtp.Password, displayName: L("DisplayName:Abp.Mailing.Smtp.Password"), description: L("Description:Abp.Mailing.Smtp.Password"), isEncrypted: true),
                new SettingDefinition(EmailSettingNames.Smtp.Domain, displayName: L("DisplayName:Abp.Mailing.Smtp.Domain"), description: L("Description:Abp.Mailing.Smtp.Domain")),
                new SettingDefinition(EmailSettingNames.Smtp.EnableSsl, "false", L("DisplayName:Abp.Mailing.Smtp.EnableSsl"), L("Description:Abp.Mailing.Smtp.EnableSsl")),
                new SettingDefinition(EmailSettingNames.Smtp.UseDefaultCredentials, "true", L("DisplayName:Abp.Mailing.Smtp.UseDefaultCredentials"), L("Description:Abp.Mailing.Smtp.UseDefaultCredentials")),
                new SettingDefinition(EmailSettingNames.DefaultFromAddress, "noreply@abp.io", L("DisplayName:Abp.Mailing.DefaultFromAddress"), L("Description:Abp.Mailing.DefaultFromAddress")),
                new SettingDefinition(EmailSettingNames.DefaultFromDisplayName, "ABP application", L("DisplayName:Abp.Mailing.DefaultFromDisplayName"), L("Description:Abp.Mailing.DefaultFromDisplayName"))
            );
        }
        private static LocalizableString L(string name)
        {
            return LocalizableString.Create<EmailingResource>(name);
        }
    }
```
and then `EmailingResource` is import `SettingResource`  by `AddVirtualJson`   or not.
I follow the Abp team's ideas

**Comment by wakuflair** [comment]
> `SettingDefinition` already have a Constructor which with `displayName`,`description` params.
Yeah I prefer using `displayName` and `description` parameteres. Once this PR is accepted, I will update my module to use them.

> and then `EmailingResource` is import `SettingResource` by `AddVirtualJson` or not.
I can't see the `SettingResource`, it's what you plan to commit?

**Comment by yinchang0626** [comment]
@wakuflair 
 I do not have plan to add SettingResource now.
In this PR. Just modify SettingDefinition to add define of displayName and description.
and going to do things is @hikalkan 's commit
```csharp
namespace Volo.Abp.Localization.Resources.AbpValidation
{
    //TODO: Move to Volo.Abp.Validation!

    [LocalizationResourceName("AbpValidation")]
    public class AbpValidationResource
    {
        
    }
}
```
but I think after moved ,will cause breaking change

**Comment by yinchang0626** [comment]
@hikalkan 
please review this commit [3ba06fe](https://github.com/abpframework/abp/pull/2392/commits/3ba06fe0227b86099f3aa718b5d3976bb0e4bbe0)
I move `AbpValidationResource` to `Volo.Abp.Validation` from `Volo.Abp.Localization`.
But still stay [`Volo.Abp.Localization.Resources.AbpValidation.AbpValidationResource`](https://github.com/abpframework/abp/blob/3ba06fe0227b86099f3aa718b5d3976bb0e4bbe0/framework/src/Volo.Abp.Localization/Volo/Abp/Localization/Resources/AbpValidation/AbpValidationResource.cs) file 
for template and module still can reference it to `AddBaseTypes` in definition of localization resource.
And mark it as `obsolete` ,maybe future version will remove it,
finally,as you say ,
I put `DisplayName:Abp.Localization.DefaultLanguage`  to new  `AbpLocalizationResource`.
Could you check it alright?thanks

**Pull request merged**